test/core/api/service/state/state_api_test.cpp

```diff
@@ -61,3 +64,60 @@ TEST(StateApiTest, GetStorage) {
   EXPECT_OUTCOME_TRUE(r1, api.getStorage("a"_buf, "B"_hash256));
   ASSERT_EQ(r1, "1"_buf);
 }
+
+/**
+ * @given state api
+ * @when get a runtime version for the given block hash
+ * @then the correct value is returned
+ */
+TEST(StateApiTest, GetRuntimeVersion) {
+  auto storage = std::make_shared<TrieStorageMock>();
+  auto block_header_repo = std::make_shared<BlockHeaderRepositoryMock>();
+  auto block_tree = std::make_shared<BlockTreeMock>();
+  auto runtime_core = std::make_shared<CoreMock>();
+
+  kagome::api::StateApiImpl api{
+      block_header_repo, storage, block_tree, runtime_core};
+
+  char const *dummy_sn = "dummy_sn";
+  char const *dummy_in = "dummy_in";
+  uint32_t const val1 = 0x101;
+  uint32_t const val2 = 0x111;
+  uint32_t const val3 = 0x201;
+
+  std::optional<kagome::primitives::BlockHash> hash1 = std::nullopt;
+  EXPECT_CALL(*runtime_core, version(hash1))
+      .WillOnce(
+          testing::Return(kagome::primitives::Version{.spec_name = dummy_sn,
+                                                      .impl_name = dummy_in,
+                                                      .authoring_version = val1,
+                                                      .spec_version = val3,
+                                                      .impl_version = val2}));
+
+  {
+    EXPECT_OUTCOME_TRUE(result, api.getRuntimeVersion(std::nullopt));
+    ASSERT_EQ(result.spec_name, dummy_sn);
```

Comment:
Version type has == and != operators, so these asserts could be replaced with a single assert